### PR TITLE
Added uint256 as input test

### PR DESCRIPTION
**Pull request by Radinyn**

### CODE REVIEW

**Review by Arcticae — APPROVED** [review]
LGTM with reservation to Arek's comments

File: tests/integration/gateway/uint256_as_input_test.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+@pytest.fixture(name="compiled_contract_filepath")
+def compiled_contract_filepath_fixture() -> Path:
+    return Path("./build/main.json")
+
+
+@pytest.mark.parametrize(
+    "constructor_input",
+    [[42, 24], {"amount": {"high": 24, "low": 42}}, {"amount": 24 << 128 + 24}],
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply tests/integration/gateway/uint256_as_input_test.py]: Maybe that's a good thing, but we need to support only one version. A contract can take many uint256 values, e.g. `foo: uint256, bar: uint256`. It's not obvious which two values represent uint256: (12, 24), 36 or 12, (24, 36).

If the official StarkNet CLI supports uint256 represented as a single value, we should also do that. Otherwise, Protostar should require 2 values, unless data transformer is used, e.g. `{ "amount": 1231231 }`.

[the remark this answers is not in the document]
Comment (Radinyn) [reply tests/integration/gateway/uint256_as_input_test.py]: StarkNet CLI supports UInt256 as two values. The point is, to add this kind input validation we need to add more logic ontop of the starknet.py data transformer which is imo not really related to this PR and honestly I'm not sure wheter its worth it since the transaction will fail instantly at StarkNet eiter way.

tldr:
Input validation based on starknet.py's data transformer is not bullet proof, but fairly good.

[the remark this answers is not in the document]
Comment (kasperski95) [reply tests/integration/gateway/uint256_as_input_test.py]: I am concerned about unexpected transformation, not validation. I think a test that checks `protostar invoke` is also needed.

[the remark this answers is not in the document]
Comment (Radinyn) [reply tests/integration/gateway/uint256_as_input_test.py]: Currently all of them gateway commands use it, perhaps I should change this test in a way that catches starknet.py's ClientError (fail at starknet from invalid input)

[the remark this answers is not in the document]
Comment (Radinyn) [reply tests/integration/gateway/uint256_as_input_test.py]: Okay, commited this approach, please let me know if it's ok.

[the remark this answers is not in the document]
Comment (kasperski95) [reply tests/integration/gateway/uint256_as_input_test.py]: Raising `ClientError` instead of `ProtostarException` is an undesired behavior, therefore there's no point of creating a test for it unless it's properly commented. 

In case of `deploy`, we apply data transformer depending on the input's type (dict or list). However, in case of `invoke` we don't do that. I suspect Starknet.py uses data transformer on a list, resulting in an inconsistent behavior across commands.

[the remark this answers is not in the document]
Comment (Radinyn) [reply tests/integration/gateway/uint256_as_input_test.py]: `ClientError` is now wrapped in `TransactionException`
